budgeteer-web-interface/src/main/java/org/wickedsource/budgeteer/service/budget/BudgetService.java

```diff
@@ -260,7 +263,19 @@ public EditBudgetData loadBudgetToEdit(long budgetId) {
      * @return the
      */
     public long saveBudget(EditBudgetData data) {
-        assert data != null;
+        boolean duplicateImportKey = budgetRepository.existsByImportKey(data.getImportKey());
```

Comment:
Unfortunately, this check is fundamentally flawed, as one cannot edit an existing budget any longer without changing the import key and the budget name.
The logic must be to check for duplicates if and only if the respective fields have changed.

Reply:
We now check if there were any changes made to the respective fields. The validation is only made if there are changes.
Additionally, we changed the existing query to only lookup into the matching project.